lib/scholar/linear/polynomial_regression.ex

```diff
@@ -0,0 +1,190 @@
+defmodule Scholar.Linear.PolynomialRegression do
+  @moduledoc """
+  Least squares polynomial regression.
+  """
+
+  import Nx.Defn
+
+  # @derive {Nx.Container, containers: [:coefficients, :intercept]}
+  # defstruct [:coefficients, :intercept]
```

Comment:
We eventually would like to return the struct with coef and intercept. So this might be useful.